src/main/java/spleetwaise/transaction/model/TransactionBookModelManager.java

```diff
@@ -87,6 +87,12 @@ public void deleteTransactionsOfPersonId(String personId) {
         transactionBook.deleteTransactionsOfPersonId(personId);
     }
 
+    @Override
+    public void deleteTransaction(Transaction transaction) {
+        transactionBook.removeTransaction(transaction);
```

Comment:
As part of defensive coding, perhaps there should be a null check here for transaction.